core/src/main/java/apoc/export/graphml/XmlGraphMLReader.java

```diff
@@ -277,16 +277,24 @@ public long parseXML(Reader input) throws XMLStreamException {
     private RelationshipType getRelationshipType(XMLEventReader reader) throws XMLStreamException {
         if (this.labels) {
             XMLEvent peek = reader.peek();
-            if (peek.isCharacters() && !(peek.asCharacters().isWhiteSpace())) {
+            boolean isChar = peek.isCharacters();
+            if (isChar && !(peek.asCharacters().isWhiteSpace())) {
                 String value = peek.asCharacters().getData();
                 String el = ":";
                 String typeRel = value.contains(el) ? value.replace(el, StringUtils.EMPTY) : value;
                 return RelationshipType.withName(typeRel.trim());
-            } else if (!peek.isEndDocument()) {
+            }
+
+            boolean notStartElementOrContainsKeyLabel = isChar
+                    || !(peek instanceof StartElement)
+                    || peek.asStartElement().getAttributes().next().getValue().equals("label");
```

Comment:
trasformalo in metodo:

```java
    private boolean containsLabelKey(XMLEvent peek) {
        final Attribute keyAttribute = peek.asStartElement().getAttributeByName(new QName("key"));
        return keyAttribute != null && keyAttribute.getValue().equals("label");
    }
```

Reply:
Questo errore è pigrizia, quando ti viene del codice scritto così devi sempre fermarti e chiederti se c'è una soluzione migliore perché così non è ben scritto; indipendentemente dal linguaggio di programmazione che usi...